crd-generator/api/src/main/java/io/fabric8/crd/generator/Resources.java

```diff
@@ -43,7 +44,10 @@ public KubernetesListBuilder global() {
    * @param decorator The decorator.
    */
   public void decorate(Decorator decorator) {
-    globalDecorators.add(decorator);
+    final boolean added = globalDecorators.add(decorator);
+    if (!added) {
+      throw new RuntimeException("Conflict adding decorator " + decorator.getClass().getName());
```

Comment:
This might not be enough, as there are some decorators that we do add multiple times

Reply:
Which ones and why?

Reply:
AddSubResourcesDecorator, because it gets added for spec & replicas path and also for status subresource.